### ISSUE TITLE
[BUG] AssetEntity.videoDuration comes back 0 even if its a video

### ISSUE THREAD

**Issue by Mayb3Nots** [CaiJingLong/flutter_photo_manager#457]
**Describe the bug**
The title says all. Recently after upgrading to null safety, when getting all media, all the videos don't have a duration anymore.

**To Reproduce**
1) Get some video media.
2) Get the videoDuration property from the AssetEntity.
3) print in console log, you will get 0 even though the video has length.
4) you can use example app ans it will give same result.

![Screenshot_20210321-195920](https://user-images.githubusercontent.com/68549184/111904160-55829400-8a80-11eb-810e-07f45b0658bf.jpg)

This is a screenshot taken using the example app. As you can see the video is a mp4 but has no duration

**Expected behavior**
Suppose to give out video's duration instead of 0

**Flutter version**
2.0.1

**Smartphone (please complete the following information):**
 - Device: Samsung A8 Star
 - OS: Andriod 9

**Comment by AlexV525** [comment]
Could you try with a non-null-safety version to verify whether it's a package issue or something else?

**Comment by AlexV525** [comment]
Also, it's noticeably that Samsung device usually ran into this issue, while another user has the same issue on SM G9730 Android 11, as reported.

**Comment by barretzhi** [comment]
iphone too, can't get video duration

**Comment by AlexV525** [comment]
I've tested with the latest version of the package today, all of them are not reproducible with this issue.

* Android Emulator (30) 
![image](https://user-images.githubusercontent.com/15884415/112090356-5aedf480-8bce-11eb-9a7b-f09345a5ded7.png)

* OnePlus 8 Pro (30)
![image](https://user-images.githubusercontent.com/15884415/112090389-6b05d400-8bce-11eb-9be7-2194337f673d.png)

* OnePlus 6T (29)
![image](https://user-images.githubusercontent.com/15884415/112090683-00a16380-8bcf-11eb-8f7b-018d0d383229.png)

* iPhone SE (14.0.1)
![image](https://user-images.githubusercontent.com/15884415/112091616-b620e680-8bd0-11eb-8ca8-063828e64fdb.png)

**Comment by Mayb3Nots** [comment]
> I've tested with the latest version of the package today, all of them are not reproducible with this issue.
> 
> * Android Emulator (30)
>   ![image](https://user-images.githubusercontent.com/15884415/112090356-5aedf480-8bce-11eb-9a7b-f09345a5ded7.png)
> * OnePlus 8 Pro (30)
>   ![image](https://user-images.githubusercontent.com/15884415/112090389-6b05d400-8bce-11eb-9be7-2194337f673d.png)
> * OnePlus 6T (29)
>   ![image](https://user-images.githubusercontent.com/15884415/112090683-00a16380-8bcf-11eb-8f7b-018d0d383229.png)
> * iPhone SE (14.0.1)
>   ![image](https://user-images.githubusercontent.com/15884415/112091616-b620e680-8bd0-11eb-8ca8-063828e64fdb.png)

Using commit reference number [ed95af8](https://github.com/CaiJingLong/flutter_photo_manager/commit/ed95af83becf547271f708e003c1adc0f3c3690d) works fine. I think. The redesign of the constructor has broken something. It is because the constructor now defaults` duration = 0` #453

**Comment by AlexV525** [comment]
Oops you're right. I've found some relative code that are really written in drunk. Will make a quick fix later.

**Comment by Mayb3Nots** [comment]
> Oops you're right. I've found some relative code that are really written in drunk. Will make a quick fix later.
> 
> TL;DR, it's not about the constructor but convert methods, which have a serious wrong code usage.

Haha it's fine, thanks for helping!
